### ISSUE TITLE
MACOS: Issue with installing pieces os and pieces package:

### ISSUE THREAD

**Issue by Emir-Salihovic** [pieces-app/support#65]
### Software

Desktop Application, Pieces OS

### Operating System

macOS

### Your Pieces OS Version

none

### Kindly describe the bug and include as much detail as possible on what you were doing so we can reproduce the bug.

Hi, I am experiencing an issue with installing pieces os and pieces package:
I am using a mac with m1 chip , so I downloaded ARM64 arch packages, my os version is macos SONOMA 14.2.1 (23C71).
Please look at the screenshots bellow for more details:
### Pieces ARM64 .pkg 
<img width="336" alt="SCR-20240126-nmhw" src="https://github.com/pieces-app/support/assets/52429514/3df80635-8ac0-42fc-bf9d-376dcb2485cf">

### Pieces OS when I download it separately
<img width="336" alt="SCR-20240126-nmqf" src="https://github.com/pieces-app/support/assets/52429514/3bc58cd6-0ee8-458d-94e9-c182c67fe61b">

### os
<img width="293" alt="SCR-20240126-nonv" src="https://github.com/pieces-app/support/assets/52429514/fc80ba8d-40b3-4c0e-b68c-f724e33093f1">

**Comment by rosie-at-pieces** [comment]
hey @Emir-Salihovic can you please try restarting your computer and seeing if that resolves this?

**Comment by Emir-Salihovic** [comment]
Hey @rosie-at-pieces  , restarting did not resolve the problem.

For more insights please take a look at the following:
1. I tried installing the product via brave, here is the output:
<img width="1439" alt="SCR-20240126-qgtx" src="https://github.com/pieces-app/support/assets/52429514/901553dc-e688-4830-8cbc-e70f2b8d1cea">
2.

https://github.com/pieces-app/support/assets/52429514/7f770db9-ae66-497c-a073-418b623d1406

3. In your tutorial on how to install pieces for mac, nothing was mentioned about apple silicon chip
[https://www.youtube.com/watch?v=1hGG98pIrvs](url)

**Comment by rosie-at-pieces** [comment]
@nathan-courtney-pieces you wanna take a look here

**Comment by nathan-courtney-pieces** [comment]
There is a few things we can try here. Typically we find that a computer restart fixes this issue, but in this case it has not. 

1.) Download the PKG installer Again
2.) Remove the quarantine flag if there is one. 
`$ xattr -d com.apple.quarantine path/to/Pieces.pkg`
3.) Right click the installer > choose "Open"
 If you get an additional prompt after the right click, accept that. 
 
 If that still doesn't work for you try restarting again. 
 
 Let me know if you still have issues after this.

**Comment by Emir-Salihovic** [comment]
Hey @nathan-courtney-pieces , please take a look
I googled to find out if the flag is present in the first place and got no flag on the .pkg file but there is one in the pieces os I downloaded, the same happens when running `xattr -d com.apple.quarantine path/to/Pieces.pkg `
also note that I am using brave browser, after this I restarted my mac once more, but the problem is still here.
<img width="1436" alt="SCR-20240126-robk" src="https://github.com/pieces-app/support/assets/52429514/6ff6f888-b69d-41c1-a422-1bef6ee36afa">

**Comment by nathan-courtney-pieces** [comment]
@Emir-Salihovic 

Thanks for trying all of that, sorry that you are still having issues. 

Lets try something else.

Can you open up Disk Utility on macOS:

1.) Go to **Finder > Applications > Utilities > Disk Utility**.

2.) Move to the top Disk Utility menu bar and select **Images > Verify**. 

3.) A file picker window should pop up. Simply select the PKG or the DMG and then click verify in the bottom right. 

If you can send a screenshot of the output of that, it would help me a lot to identify the issue.

**Comment by Emir-Salihovic** [comment]
@nathan-courtney-pieces hope this is helpful to you.

https://github.com/pieces-app/support/assets/52429514/f6b25b62-8fdb-474c-9af3-0ed1d8ff7aef

**Comment by nathan-courtney-pieces** [comment]
@Emir-Salihovic 

Thanks for trying that... Unfortunately it doesn't help me a whole lot, I was hoping for something other than "image not recognized".

I am not giving up yet!

Lets try something else, but first, may I ask what browser are you currently using to download these archives? 

Let's make sure that you allow apps downloaded from outside the App Store. Under Security and Privacy within your settings, make sure you check the box that allows other applications to be run from identified developers.

<img width="721" alt="image" src="https://github.com/pieces-app/support/assets/46496645/4b11d8f1-f96e-4f13-9c7b-4e172b3cf391">

If you previously had this disabled and enabled it just now, I would recommend restarting your computer as well.

Go ahead and delete all the PKG's and DMG's that you already downloaded.

Let's try to download the file through `wget` and open after the fact. 

`cd ~/Downloads`

`wget https://builds.pieces.app/stages/production/os_server/dmg-arm64/download -O Pieces_arm64.dmg`

After it finishes downloading, try and right click to open the archive. 

Let me know if you are still seeing that error message. I appreciate your engagement and patience trying to get this to work on your machine. I want to get to the bottom of this problem as much as you do!

**Comment by Emir-Salihovic** [comment]
@nathan-courtney-pieces thank you! Here are the results:
1. I'm using brave browser.
2. The option for apps besides app store was already enabled, anyways I restarted the computer as you proposed.
<img width="720" alt="SCR-20240129-sknr" src="https://github.com/pieces-app/support/assets/52429514/0ffdbf35-7dde-44e8-b0d3-ccedc6b4148f">

3. I successfully downloaded the new package using wget and deleted the old ones.
4. After trying to open it here is the message I got.
<img width="1440" alt="SCR-20240129-spnq" src="https://github.com/pieces-app/support/assets/52429514/2fcf09a8-f341-4155-9702-52a2d6eabc25">
5. After that I used the disk utility again to verify the image, and there also the message as before was "image not recognized"

**Comment by nathan-courtney-pieces** [comment]
@Emir-Salihovic 

Well darn, thanks for trying that. 

I want to confirm that our server isn't having any issues. 

Below you will find a link that directly links to our DMG file within cloud storage. I am wondering if it's an issue with our proxy. 

Delete what you have downloaded before, and try to download and open from this link. You may restart if you want, but I do not think it would be necessary unless you wanted to give it a go. 

https://storage.googleapis.com/app-releases-production/os_server/56gha/Pieces-OS.dmg

Let me know if that works for you or not

**Comment by Emir-Salihovic** [comment]
Hey @nathan-courtney-pieces this fixed the issue with pieces OS , also I would like to mention that for this time I used safari for download , so please take that also into consideration, but is this the apple silicon version or the intel one? The icon of pieces os appeared in my menu bar but I cannot open it, as soon as I hover the icon it starts the mac loading indicator.

**Comment by nathan-courtney-pieces** [comment]
@Emir-Salihovic 

That is good news! I do believe it is an issue with our server if you are able to run it from that. I will have to look into this and see what I can do to prevent this from happening. 

I believe that I had given you the intel version. Although it technically should work, you may still have some problems with that. 

Here is a direct link to the PKG installer for ARM macOS that will get you up and running with everything you need:

https://storage.googleapis.com/app-releases-production/macos_packaging/62gha/Pieces-arm64.pkg

I would move the previously installed Pieces OS to the trash first. 

Let me know if that works for you. If it does then that is fantastic. I will be modifying our server today to see if I can get this issue resolved. You have been much help, if you are willing to help me out a little later and test downloading from our website I would appreciate it to confirm that the issue is going to be resolved for anyone with a similar problem.

**Comment by Emir-Salihovic** [comment]
@nathan-courtney-pieces it worked! 🥳 I can help you of course, please let me know when the server is ready.
Just one tiny bug that I noticed, every time the product gets installed for the first time pieces os freezes , so I needed to force quit it inside activity monitor and only after the second launch it worked properly!

**Comment by nathan-courtney-pieces** [comment]
@Emir-Salihovic 

That is great news! 

Pieces OS has a big boot up process on first launch that eventually gets resolved, which may be the cause of hanging on your machine. Entirely separate issue that we will have to optimize :) 

If you wouldn't mind, try to see if you can download and open the PKG from this link:

https://builds.staging.pieces.app/stages/production/macos_packaging/pkg/download

You do not need to actually run the installer, I simply want to know if you are able to open it without any of the original issues. Thank you so much for taking time to help me out

**Comment by Emir-Salihovic** [comment]
@nathan-courtney-pieces this is the problem that occurred with this package.
Are you sure this is the arm64 arch?
<img width="1440" alt="SCR-20240130-kjkv" src="https://github.com/pieces-app/support/assets/52429514/bc6fcead-b1fa-4eb1-932f-fcbaf5070d79">

**Comment by nathan-courtney-pieces** [comment]
@Emir-Salihovic 

My bad, I did it again! I am working on Intel so I am sorry I keep sending you the wrong one.

Here is the appropriate link

https://builds.staging.pieces.app/stages/production/macos_packaging/pkg-arm64/download

You may notice that you will now receive download progress as opposed to an indeterminate file download.

Let me know if this works for you, again, you don't _have to_ run the installer, but if you do it will just overwrite the apps and relaunch. It won't affect any data that was generated from the application i.e. anything saved within it. I just want to make sure you don't receive any errors related to digital signatures or corrupted files.

**Comment by Emir-Salihovic** [comment]
@nathan-courtney-pieces this worked again! :)

**Comment by nathan-courtney-pieces** [comment]
@Emir-Salihovic 

That is great news! 

You do not need to change anything. You should currently have the latest production release and the apps will check for updates themselves when a new version is pushed. You should be on Pieces OS 7.2.7 and Pieces for Developers 2.9.3.

You can manually check for updates by clicking on the "P" icon within your menu bar at the top for Pieces OS. If there is an update available, you will see a button to download and restart within that menu that pops down, and Pieces for Developers desktop application can be manually checked through the settings menu by clicking on your profile icon. There will be a button in there that allows you to manually check. You will see a oval looking button in the top right of the app that will tell you if an update is available regardless if you manually check periodically or not. 

Thanks for all your help! 

The issue should now be resolved @rosie-at-pieces
